rfc/2022/2022-07-13 - Platform Wide Documentation vs Component Specific Documentation Locations.md

```diff
@@ -0,0 +1,88 @@
+# RFC: Platform-wide VFS developer documentation will be viewed on the VA Platform Website; component-specific technical documentation will be viewed on the Console UI
+
+<!--
+The title is what you want comments on. Use the active voice in a future tense.
+Example:
+    - The website will be built using the XZY framework
+-->
+
+* Comment Deadline: `2022-08-31`
+* Team Crew and Name: [Platform/Spike Teams/Console UI](https://github.com/orgs/department-of-veterans-affairs/teams/platform-pst-console-ui)
+* Authors:
+  * [Thomas Ferrin](https://github.com/TPurple)
+  * [John Chapin](https://github.com/jchapin)
+  * [Susanna Kline](https://github.com/sdk-irl)
+* [Original RFC Pull Request](https://github.com/department-of-veterans-affairs/va.gov-platform-architecture/pull/33)
+
+## Background
+
+The Platform currently has two centralized locations for documentation:
+* The (Platform website)[https://depo-platform-documentation.scrollhelp.site/index.html], which uses (Scroll Viewport)[https://www.k15t.com/software/scroll-viewport/scroll-viewport-for-confluence-cloud] to display specific content hosted in Confluence. This website hosts the following types of content:
+    * **“Getting started”** content for onboarding VFS team members to work on the Platform
+    * **“Developer docs”** content about writing and submitting code for review
+    * **“Research and design”** content for adhering to the Platform’s guidance for research and design
+    * **“Analytics and insights”** content, such as for analytics, KPI dashboards, and monitoring
+    * **“Collaboration cycle”** content
+    * **“Support”** policies and incident management
+    * **“User guides, how-to instructions, tutorials, or reference documents for VFS teams”** ((source)[https://vfs.atlassian.net/wiki/spaces/CUI/pages/2186543112/RFC+DRAFT+Documentation+Locations+-+Console+UI])
+* (Operational documentation)[https://vfs.atlassian.net/wiki/spaces/AP/pages/1287782439] throughout Confluence, which includes internal content such as “(meeting notes, roles and responsibilities documentation, decision documentation, [and] initiative outlines.)[https://vfs.atlassian.net/wiki/spaces/AP/pages/1287782439]”
+
+While Confluence and the Platform website contain helpful documentation for VFS (and Platform) teams, a lot of documentation—specifically technical documentation for a software component, service, or infrastructure resource—is found throughout various GitHub repositories. This documentation is vital to developers trying to troubleshoot or set up software dependencies, but it is much harder to find. Developers typically find it by asking around, receiving a few different answers, and sometimes finding different versions of the documentation. When they find the documentation, it is often difficult to know whether it is complete or up to date. Additionally, some VFS teams rely on systems external to the VA and have had difficulty finding or maintaining documentation about these systems.
```

Comment:
Could you provide some definition of what you mean by a "software component"? Or use a different word or phrase that might be clearer to those of us who have not worked with Backstage? I understand that Backstage uses the word "component" in a specific way, but it is not a common usage of this word and I'm struggling a little to understand exactly what it means here. It's difficult, I think, because depending on the context the word "component" is used to mean several different things, so adding one more definition almost feels like overload.

Reply:
In general, do we mean to use CUI as a viewing (only) platform the intended documentation or  would edits and maintenance have to happen within CUI?

Also, I second Chris' concerns.
1. It should be easily sharable, especially in  Slack, Google Meet, and Zoom.
2. (Search). More than any other feature, search adds the most leverage when sifting through docs. I will specify one feature I would like to see implemented: phrase search ( or better yet boolean search) as to the aggregate of all words entered in the search field.
3. (Do we need to be logged in) While it's true that having  documentation in a central location is nice, it's also important to have least amount of hurdles to access as possible.

Reply:
We're struggling with wording a bit here. Backstage has multiple types of "entities," including users, groups, APIs, websites, etc. We were trying to find a term that refers to software entities (not people entities). So "software component" here refers to any website or part of a website, as well as any API or service. I think we would also include infrastructure to some extent. So this includes things like forms, auth-N/auth-Z mechanisms, APIs, microservices—chunks of software that make up VA.org. Does that clear it up?

Reply:
Kind of... I think the problem is that we already use the word "component" to mean something different on the Platform (design system component, application component, etc), there is significant potential here for confusion and misunderstanding.

Reply:
@k80bowman we're writing documentation about the Console that will go on the Platform Website. I think we did a fairly good job at clarifying this issue in that documentation. You can check out the [draft on Confluence](https://vfs.atlassian.net/wiki/spaces/PWD/pages/2311585838/The+Console+Catalog#Entity-Types-in-the-Catalog)